.github/workflows/latest-release.yaml

```diff
@@ -19,16 +14,38 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  files-changed:
+    name: Detect what files changed
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    outputs:
+      nimbus: ${{ steps.filter.outputs.nimbus}}
+      adapters: ${{ steps.filter.outputs.adapters }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v3.0.2
+        id: filter
+        with:
+          filters: |
+            nimbus:
+              - 'cmd/**'
```

Comment:
What if the dockerfile changes ? Will this filter be able to detect it ?

Reply:
> What if the dockerfile changes, or the Makefile changes ? 

We don't need to worry about these because this action only runs after the [pr-checks](https://github.com/5GSEC/nimbus/blob/main/.github/workflows/pr-checks.yaml) workflow successfully completes all its checks and when changes are merged to the main branch.  Therefore, any issues in the Dockerfile or Makefile will be caught by the pr-checks workflow before the merge (e.g., [pr-checks](https://github.com/5GSEC/nimbus/pull/119/files#diff-7a6a4578fe04c4fc86b28e2dc9672e48d1a9f721315e540b90bac0ef12263053R107))

> Will this filter be able to detect it ?

No.

Reply:
Suppose there are no issues with the Dockerfile, but.. , there are changes in the Dockerfile which modify the build such as a different COPY or USER directive. In this case, we might need to generate a new image ?

Reply:
Same comment for Makefile also. There might be changes to the Makefile such as go build flags which change the build